src/plugins/hyperbee/hyperbee.h

```diff
@@ -0,0 +1,183 @@
+ *                                                                         *
+ * If you have received a written license agreement or contract for        *
+ * Covered Software stating terms other than these, you may choose to use  *
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef HYPERBEE_H
+#define HYPERBEE_H
+
+#include <vector>
+#include <memory>
+#include <set>
+#include <glib.h>
+#include <libdrakvuf/filesystem.hpp>
+#include "plugins/private.h"
+#include "plugins/plugins_ex.h"
+
+/**
+ *This flag can be used to discard any command line logs except for malware detections (or those, the malware flag is true).
+ */
+#define LOG_ALWAYS 0
+
+/**
+ * Sets the default option for malware
+ */
+#define MALWARE_DEFAULT 1
+
+/**
+ * Dumps VAD or Pages
+ */
+#define DUMP_VAD 0
+
+/**
+ * A special flag to activate the permanent dumping manually
+ */
+#define DUMP_ALWAYS 1
+
+/**
+ * Ignores dlls / exes that are within System32 or SysWOW64
+ */
+#define IGNORE_SYSTEM_DLL 1
```

Comment:
These defines sound like they should be configurable (maybe in configure.ac)

Reply:
You mean via command line args or is there another way to use configure.ac for configuration purposes?

Reply:
Adding command line flags would make a mess - I'd add just `#defines` in configure.ac, then you can select options when you `configure` the project. Refer to https://github.com/tklengyel/drakvuf/blob/master/configure.ac#L211-L219 as an example.

Reply:
I've been playing around with configure.ac for around 90min and I don't get a step forward. I added the line 
```
AC_DEFINE_UNQUOTED(HYPERBEE_IGNORE_SYSTEM_DLL, 1, "")
```
inside the if condition (as you can see) and tried to access that within my hyperbee.h but I had no success.

```
AC_ARG_ENABLE([plugin_hyperbee],
  [AS_HELP_STRING([--disable-plugin-hyperbee],
    [Enable the hyperbee plugin @<:@yes@:>@])],
  [plugin_hyperbee="$enableval"],
  [plugin_hyperbee="yes"])
AM_CONDITIONAL([PLUGIN_HYPERBEE], [test x$plugin_hyperbee = xyes])
if test x$plugin_hyperbee = xyes; then
  AC_DEFINE_UNQUOTED(ENABLE_PLUGIN_HYPERBEE, 1, "")
  AC_DEFINE_UNQUOTED(HYPERBEE_IGNORE_SYSTEM_DLL, 1, "")
fi
```
Do you have a MWE how to define and access such new variable?

Reply:
@tklengyel knows autotools much better than me, maybe he could help. From a quick search (https://github.com/tklengyel/drakvuf/search?q=ENABLE_PLUGIN_PROCDUMP) I think it should work. I would add separate case for each parameter tho.
```
AC_ARG_ENABLE([plugin_hyperbee],
  [AS_HELP_STRING([--disable-plugin-hyperbee],
    [Enable the hyperbee plugin @<:@yes@:>@])],
  [plugin_hyperbee="$enableval"],
  [plugin_hyperbee="yes"])
AM_CONDITIONAL([PLUGIN_HYPERBEE], [test x$plugin_hyperbee = xyes])
if test x$plugin_hyperbee = xyes; then
  AC_DEFINE_UNQUOTED(ENABLE_PLUGIN_HYPERBEE, 1, "")
  AC_ARG_ENABLE([hyperbee_ignore_system_dll],
    [AS_HELP_STRING([--disable-plugin-hyperbee],
      [description here @<:@yes@:>@])],
    [hyperbee_ignore_system_dll="$enableval"],
    [hyperbee_ignore_system_dll="yes"])
...
```

Reply:
I don't think these should be compile-time flags, they look to me like runtime-configuration flags, so these would just go into `hyperbee_config_struct` below.

Reply:
Implementing rn

Reply:
Implemented & Tested. Maybe one can have an additional look on style / drakvuf rules.